core/imgread/common.cpp

```diff
@@ -138,158 +152,122 @@ Disc* OpenDisc(const wchar* fn)
 	return rv;
 }
 
-bool InitDrive_(wchar* fn)
+bool InitDrive_(const wchar* const fn)
 {
-	TermDrive();
+    TermDrive();
 
-	//try all drivers
-	disc = OpenDisc(fn);
+    //try all drivers
+    disc = OpenDisc(fn);
 
-	if (disc!=0)
-	{
-		printf("gdrom: Opened image \"%s\"\n",fn);
-		NullDriveDiscType=Busy;
+    if (disc)
+    {
+        printf("gdrom: Opened image \"%s\"\n", fn);
+        NullDriveDiscType = Busy;
 #ifndef NOT_REICAST
-		libCore_gdrom_disc_change();
+        libCore_gdrom_disc_change();
 #endif
-//		Sleep(400); //busy for a bit // what, really ?
-		return true;
-	}
-	else
-	{
-		printf("gdrom: Failed to open image \"%s\"\n",fn);
-		NullDriveDiscType=NoDisk; //no disc :)
-	}
-	return false;
+        return true;
+    }
+    else
+    {
+        printf("gdrom: Failed to open image \"%s\"\n", fn);
+        NullDriveDiscType = NoDisk; //no disc :)
+        return false;
+    }
 }
 
 #ifndef NOT_REICAST
-bool InitDrive(u32 fileflags)
+bool InitDrive(u32 fileflags/*=0*/)
 {
 	if (settings.imgread.LoadDefaultImage)
 	{
-		printf("Loading default image \"%s\"\n",settings.imgread.DefaultImage);
-		if (!InitDrive_(settings.imgread.DefaultImage))
+		printf("Loading default image \"%s\"\n", settings.imgread.DefaultImage.c_str());
+		if (!InitDrive_(settings.imgread.DefaultImage.c_str()))
 		{
-			msgboxf("Default image \"%s\" failed to load",MBX_ICONERROR);
+			msgboxf("Default image \"%s\" failed to load", MBX_ICONERROR);
 			return false;
 		}
 		else
 			return true;
 	}
 
-	// FIXME: Data loss if buffer is too small
-	wchar fn[512];
-	strncpy(fn,settings.imgread.LastImage, sizeof(fn));
-	fn[sizeof(fn) - 1] = '\0';
-
+	std::string diskImageFileName{ settings.imgread.LastImage };
```

Comment:
Should be able to use `=` instead of `{ ... }` for initialization

Reply:
kk... why braced-init is the problem?